components/List/examples/MDCSelfSizingStereoCellExample.m

```diff
@@ -0,0 +1,163 @@
+-(void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  [self positionCollectionView];
+}
+
+- (void)createDataSource {
+  self.numberOfCells = 100;
+  NSMutableArray *randomStrings = [[NSMutableArray alloc] initWithCapacity:self.numberOfCells];
+  for (NSInteger i = 0; i < self.numberOfCells; i++) {
+    [randomStrings addObject:[self generateRandomString]];
+  }
+  self.randomStrings = [randomStrings copy];
+}
+
+- (void)createCollectionView {
+  self.collectionViewLayout = [[UICollectionViewFlowLayout alloc] init];
+  self.collectionViewLayout.estimatedItemSize =
+      CGSizeMake(self.collectionView.bounds.size.width, kArbitraryCellHeight);
+  self.collectionViewLayout.minimumInteritemSpacing = 1;
+  self.collectionViewLayout.minimumLineSpacing = 0;
+  self.collectionView = [[UICollectionView alloc] initWithFrame:self.view.bounds
+                                           collectionViewLayout:self.collectionViewLayout];
+  self.collectionView.backgroundColor = [UIColor whiteColor];
+  [self.collectionView registerClass:[MDCSelfSizingStereoCell class]
+          forCellWithReuseIdentifier:kSelfSizingStereoCellIdentifier];
+  self.collectionView.delegate = self;
+  self.collectionView.dataSource = self;
+  [self.view addSubview:self.collectionView];
+}
+
+- (void)positionCollectionView {
+  CGFloat originX = self.view.bounds.origin.x;
+  CGFloat originY = self.view.bounds.origin.y;
+  CGFloat width = self.view.bounds.size.width;
+  CGFloat height = self.view.bounds.size.height;
+  if (@available(iOS 11.0, *)) {
+    originX += self.view.safeAreaInsets.left;
+    originY += self.view.safeAreaInsets.top;
+    width -= (self.view.safeAreaInsets.left + self.view.safeAreaInsets.right);
+    height -= (self.view.safeAreaInsets.top + self.view.safeAreaInsets.bottom);
+  }
+  CGRect frame = CGRectMake(originX, originY, width, height);
+  self.collectionView.frame = frame;
+  self.collectionViewLayout.estimatedItemSize =
+      CGSizeMake(self.collectionView.bounds.size.width, kArbitraryCellHeight);
+  [self.collectionViewLayout invalidateLayout];
+}
+
+-(NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
+  return 1;
+}
+
+-(NSInteger)collectionView:(UICollectionView *)collectionView
+    numberOfItemsInSection:(NSInteger)section {
+  return self.numberOfCells;
+}
+
+-(UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
+                 cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+  MDCSelfSizingStereoCell *cell =
+      [collectionView dequeueReusableCellWithReuseIdentifier:kSelfSizingStereoCellIdentifier
+                                                forIndexPath:indexPath];
+  cell.titleLabel.text = self.randomStrings[indexPath.item];
+  cell.detailLabel.text = self.randomStrings[(indexPath.item + 1) % self.randomStrings.count];
+  cell.titleLabel.textColor = [UIColor darkGrayColor];
+  cell.detailLabel.textColor = [UIColor darkGrayColor];
+  cell.titleLabel.numberOfLines = 0;
```

Comment:
if the cell is self-sizing, is it worth having the labels that can expand be `numberOfLines = 0` by default?

Reply:
If it's the default value of 1 the cell's labels truncate. I wanted them to wrap so the cell can expand vertically.

Reply:
definitely, i was wondering about the implementation itself, is it numberOfLines=0 as a default state? and if it is we can remove this line, otherwise, is there a reason why it isn't?

Reply:
I understand now, we spoke offline. I am setting the `numberOfLines` both in the cell and in the example. This should only be done in one place. @yarneo and I agree that it should be in the cell.

Reply:
To re-confirm a discussion we’d had about this as well: this will make the cell expand without bound by default, which is different from most UIKit cells’ default behavior. We need to either:

1. Very clearly document this difference in default behavior.
2. Align with UIKit.

My vote is for #2 because it’s less documentation for us to maintain.

Reply:
To re-confirm a discussion we’d had about this as well: this will make the cell expand without bound by default, which is different from most UIKit cells’ default behavior. We need to either:

1. Very clearly document this difference in default behavior.
2. Align with UIKit.

My vote is for #2 because it’s less documentation for us to maintain.

Reply:
makes sense. The place I was coming from is that if in fact the point of this cell is to be self-sizing, we would probably want our users to opt-out if they want to limit their size rather than opt-in to a self-sizing cell. But if we are preferring to go with expected defaults of UIKit then yeah it should be left to the example.

Reply:
This conversation makes me think we should document it either way since we seem to have different intuitions about where these lines should go